Serializer/Jms/Handler/ResourceHandler.php

```diff
@@ -0,0 +1,106 @@
+
+namespace Symfony\Cmf\Bundle\ResourceRestBundle\Serializer\Jms\Handler;
+
+use JMS\Serializer\Handler\SubscribingHandlerInterface;
+use JMS\Serializer\GraphNavigator;
+use JMS\Serializer\JsonSerializationVisitor;
+use JMS\Serializer\Context;
+use PHPCR\NodeInterface;
+use Puli\Repository\Api\Resource\Resource;
+use Symfony\Cmf\Component\Resource\RepositoryRegistryInterface;
+use Symfony\Cmf\Bundle\ResourceRestBundle\Registry\PayloadAliasRegistry;
+use Symfony\Cmf\Bundle\ResourceRestBundle\Registry\EnhancerRegistry;
+use Symfony\Cmf\Component\Resource\Repository\Resource\CmfResource;
+
+/**
+ * Handle PHPCR resource serialization
+ *
+ * @author Daniel Leech <daniel@dantleech.com>
+ */
+class ResourceHandler implements SubscribingHandlerInterface
+{
+    private $registry;
+    private $payloadAliasRegistry;
+    private $enhancerRegistry;
+
+    public function __construct(
+        RepositoryRegistryInterface $registry,
+        PayloadAliasRegistry $payloadAliasRegistry,
+        EnhancerRegistry $enhancerRegistry
+    ) {
+        $this->registry = $registry;
+        $this->payloadAliasRegistry = $payloadAliasRegistry;
+        $this->enhancerRegistry = $enhancerRegistry;
+    }
+
+    public static function getSubscribingMethods()
+    {
+        return array(
+            array(
+                'event' => GraphNavigator::DIRECTION_SERIALIZATION,
+                'format' => 'json',
+                'type' => 'Puli\Repository\Api\Resource\Resource',
+                'method' => 'serializeResource',
+            ),
+        );
+    }
+
+    /**
+     * @param JsonSerializationVisitor $visitor
+     * @param NodeInterface $resourceInterface
+     * @param array $type
+     * @param Context $context
+     */
+    public function serializeResource(
+        JsonSerializationVisitor $visitor,
+        Resource $resource,
+        array $type,
+        Context $context
+    ) {
+
+        $data = $this->doSerializeResource($resource);
+        $context->accept($data);
+    }
+
+    private function doSerializeResource(Resource $resource)
+    {
+        $data = array();
+        $repositoryAlias = $this->registry->getRepositoryAlias($resource->getRepository());
+
+        $data['repository_alias'] = $repositoryAlias;
+        $data['repository_type'] = $this->registry->getRepositoryType($resource->getRepository());
+        $data['payload_alias'] = $this->payloadAliasRegistry->getPayloadAlias($resource);
+
+        if ($resource instanceof CmfResource) {
+            $data['payload_type'] = $resource->getPayloadType();
+        }
+
+        $data['path'] = $resource->getPath();
+        $data['repository_path'] = $resource->getRepositoryPath();
+        $children = $resource->listChildren();
+
+        $enhancers = $this->enhancerRegistry->getEnhancers($repositoryAlias);
+
+        $children = array();
+        foreach ($resource->listChildren() as $name => $resource) {
+            $children[$name] = $this->doSerializeResource($resource);
```

Comment:
Instead of expecting the serializer to serialize its own type, we just recursively handle the serialization of the child type ourselves.